bson/_cbsonmodule.c

```diff
@@ -1760,6 +1763,9 @@ int write_dict(PyObject* self, buffer_t buffer,
                             if (msg) {
                                 // Prepend doc to the existing message
                                 PyObject *dict_str = PyObject_Str(dict);
+                                if (dict_str ==  NULL) {
+                                    return 0;
```

Comment:
Same leak comment here except we'll also leak more variables, eg msg.

Reply:
Fixed